ingest/src/main/scala/hydra.ingest/http/IngestionEndpoint.scala

```diff
@@ -99,6 +155,19 @@ class IngestionEndpoint[F[_]: Futurable](
     )
   }
 
+
+  private def pubV2 = parameter("correlationId" ?)  { cIdOpt =>
+    extractRequest { req =>
+      extractExecutionContext { implicit ec =>
+        onSuccess(createRequest[HttpRequest](cIdOpt.getOrElse(cId), req)) { hydraRequest =>
+          extractUnmatchedPath { topic =>
+            complete(topic.toString.replace("/",""))
+          }
```

Comment:
I don't love using `extractUnmatchedPath` here, but I could be convinced. The thing that I don't love is that it requires me to know the context of where this method is getting called from to understand it. It also makes to possible to call this method incorrectly by not matching on all the right parts of the path prior to calling it.

My 2 cents is that it would be preferable to extract the `topicName` prior to calling this method and then passing it as a parameter. What are your thoughts?

Reply:
Happy to change it. I agree passing it in as a parameter works great.